### PR TITLE
mobile: Fix flaky ClientIntegrationTest

### PR DESCRIPTION
**Pull request by fredyw**
This PR removes the test expectation that the `end_stream` always contains 0 bytes since this behavior is not guaranteed causing test flakiness. This is similar to https://github.com/envoyproxy/envoy/pull/34252. See also https://github.com/envoyproxy/envoy-mobile/issues/1393.

Fixes https://github.com/envoyproxy/envoy/issues/36074.

Risk Level: low (test only)
Testing: `bazel test //test/common/integration:client_integration_test --runs_per_test=200`
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: mobile/test/common/integration/client_integration_test.cc [hunk 1]
```diff
@@ -206,13 +206,7 @@ void ClientIntegrationTest::basicTest() {
...
 
   EnvoyStreamCallbacks stream_callbacks = createDefaultStreamCallbacks();
-  stream_callbacks.on_data_ = [this](const Buffer::Instance& buffer, uint64_t length,
-                                     bool end_stream, envoy_stream_intel) {
-    if (end_stream) {
-      std::string response_body(length, ' ');
-      buffer.copyOut(0, length, response_body.data());
-      EXPECT_EQ(response_body, "");
-    }
+  stream_callbacks.on_data_ = [this](const Buffer::Instance&, uint64_t, bool, envoy_stream_intel) {
```
Comment (abeyad) [inline mobile/test/common/integration/client_integration_test.cc:209]: so we aren't checking response body anymore?

Comment (fredyw) [reply mobile/test/common/integration/client_integration_test.cc:209]: We have never checked the response body. The existing test expectation is only checking if it's empty at the end of the stream.

**Review by ravenblackx — APPROVED** [review]

**Review by abeyad — APPROVED** [review]

**Pull request merged**